endgame_postprocessing/model_wrappers/sch/run_sch.py

```diff
@@ -153,9 +164,85 @@ def canonicalise_raw_sth_results(input_dir, output_dir, worm_directories):
             output_dir, file_info, all_worms_canonical
         )
 
+def _check_iu_in_all_folders(worm_iu_info):
+    info = {}
+    unique_worms = set()
+    for worm, _, iu, scenario in worm_iu_info:
+        unique_worms.add(worm)
+        if scenario not in info:
+            info[scenario] = {}
+        if iu not in info[scenario]:
+            info[scenario][iu] = {}
+        if worm not in info[scenario][iu]:
+            info[scenario][iu][worm] = 0
+        info[scenario][iu][worm] += 1
+        if info[scenario][iu][worm] > 1:
+            raise Exception(
+                f"IU {iu} found multiple times for {worm} in scenario {scenario}."
+            )
+
+    for scenario in info.keys():
+        for iu in info[scenario].keys():
+            iu_worms = info[scenario][iu].keys()
+            for worm in unique_worms:
+                if worm != "haematobium" and worm not in iu_worms:
+                    raise Exception(
+                        f"IU {iu} not present for {worm}."
+                    )
+
+def canonicalise_raw_sch_results(
+    input_dir,
+    output_dir,
+    all_worm=False,
```

Comment:
I read "any worm" in the `model_wrappers/sch/README.md`, but `all_worm` here. I think this is referring to the same thing? Also it looks to me that `all_worm = True` could mean only processing a subset of worm dirs, not necessarily *all* of them, correct?

Ah - and I think the README file mentionned above should be updated to describe the new behaviour you implement in this PR.

Reply:
Good points for the naming, and updating the README! I shall update those!